common/core/desktop/src/kmx/kmx_processor.cpp

```diff
@@ -2,188 +2,184 @@
 #include "state.hpp"
 #include "kmx/kmx_processor.hpp"
 
-namespace km {
-  namespace kbp
-  {
+using namespace km::kbp;
+using namespace kmx;
 
-    km_kbp_status kmx_processor::validate() const {
-      return _valid ? KM_KBP_STATUS_OK : KM_KBP_STATUS_INVALID_KEYBOARD;
-    }
+km_kbp_status kmx_processor::validate() const {
+  return _valid ? KM_KBP_STATUS_OK : KM_KBP_STATUS_INVALID_KEYBOARD;
+}
 
-    kmx_processor::kmx_processor(kbp::path p)
-    {
-      p.replace_extension(".kmx");
-      _valid = bool(_kmx.Load(p.c_str()));
+kmx_processor::kmx_processor(kbp::path p)
+{
+  p.replace_extension(".kmx");
+  _valid = bool(_kmx.Load(p.c_str()));
 
-      if (!_valid)
-        return;
+  if (!_valid)
+    return;
 
-      keyboard_attributes::options_store defaults;
-      _kmx.GetOptions()->Init(defaults);
+  keyboard_attributes::options_store defaults;
+  _kmx.GetOptions()->Init(defaults);
 
-      for (auto const & opt: defaults)
-      {
-        if (!opt.empty() && opt.scope == KM_KBP_OPT_KEYBOARD  )
-          persisted_store()[opt.key] = opt.value;
+  for (auto const & opt: defaults)
+  {
+    if (!opt.empty() && opt.scope == KM_KBP_OPT_KEYBOARD  )
+      persisted_store()[opt.key] = opt.value;
+  }
+  // Fill out attributes
+  auto v = _kmx.GetKeyboard()->Keyboard->version;
+  auto vs = std::to_string(v >> 16) + "." + std::to_string(v & 0xffff);
+
+  _attributes = keyboard_attributes(static_cast<std::u16string>(p.stem()),
+                  std::u16string(vs.begin(), vs.end()), p.parent(), defaults);
+}
+
+char16_t const * kmx_processor::lookup_option(km_kbp_option_scope scope, std::u16string const & key) const
+{
+  char16_t const * pValue = nullptr;
+  switch(scope)
+  {
+    case KM_KBP_OPT_KEYBOARD:
+      pValue = _kmx.GetOptions()->LookUp(key);
+      break;
+    case KM_KBP_OPT_ENVIRONMENT:
+      pValue = _kmx.GetEnvironment()->LookUp(key);
+      break;
+    default:
+      break;
+  }
+
+  return pValue ? pValue : nullptr;
+}
+
+option kmx_processor::update_option(km_kbp_option_scope scope, std::u16string const & key, std::u16string const & value)
+{
+  switch(scope) {
+    case KM_KBP_OPT_KEYBOARD:
+      _kmx.GetOptions()->Set(key, value);
+      persisted_store()[key] = value;
+      break;
+    case KM_KBP_OPT_ENVIRONMENT:
+      _kmx.GetEnvironment()->Set(key, value);
+      break;
+    default:
+      return option();
+      break;
+  }
+
+  return option(scope, key, value);
+}
+
+km_kbp_status kmx_processor::process_event(km_kbp_state *state,
+                                            km_kbp_virtual_key vk,
+                                            uint16_t modifier_state,
+                                            uint8_t is_key_down) {
```

Comment:
```suggestion
km_kbp_status kmx_processor::process_event(
  km_kbp_state *state,
  km_kbp_virtual_key vk,
  uint16_t modifier_state,
  uint8_t is_key_down
) {
```

It might take a little getting used to, but I think we should use this style going forward for long function signatures...